### PR TITLE
Update SVTR Tiny Model

### PR DESCRIPTION
**Pull request by zhtmike**
1. Update SVTR training data, to align with official training dataset.
2. Update LMDB dataset generator, support 1. dropping instance with zero text; 2. dropping instance which text length is larger than the maximum number the model can handle (especially for CTC alignment); 3. label standardization (NFKD)
3. Fix the SVTR augmentations, now all random variables should be randomized in `__call__` instead of `__init__`
4. Update SVTR Tiny model **accuracy: 89.02% -> 90.23%, FPS: 2968 -> 4560**
5. Clear lot of warnings when model is running on Mindspore 2.0, including legacy warning of API change of `nn.Dropout` and `ms_function`
6. Fix SVTR convolutional kernel and support dropping positional encoding in SVTR backbone

Thank you for your contribution to the MindOCR repo.
Before submitting this PR, please make sure:

- [x] You have read the [Contributing Guidelines on pull requests](https://github.com/mindspore-lab/mindcv/blob/main/CONTRIBUTING.md)
- [x] Your code builds clean without any errors or warnings
- [x] You are using approved terminology
- [ ] You have added unit tests

## Motivation

(Write your motivation for proposed changes here.)

## Test Plan

(How should this PR be tested? Do you require special setup to run the test or repro the fixed bug?)

## Related Issues and PRs

(Is this PR part of a group of changes? Link the other relevant PRs and Issues here. Use https://help.github.com/en/articles/closing-issues-using-keywords for help on GitHub syntax)

### CODE REVIEW

File: mindocr/data/transforms/svtr_transform.py [hunk 1]
```diff
@@ -242,59 +239,66 @@ def __init__(self, factor=4, base_size=(128, 512)):
...
         self.base_h, self.base_w = base_size[:2]
 
     def __call__(self, img):
-        if self.factor == 0:
+        if isinstance(self.factor, numbers.Number):
+            factor = round(sample_uniform(0, self.factor))
+        elif isinstance(self.factor, (tuple, list)) and len(self.factor) == 2:
+            factor = round(sample_uniform(self.factor[0], self.factor[1]))
+        else:
+            raise RuntimeError("factor must be number or list with length 2")
```
Comment (hadipash) [inline mindocr/data/transforms/svtr_transform.py:252]: How about using lambda functions inside `__init__` to remove data type checking inside `__call__`? Just like here: https://github.com/mindspore-lab/mindocr/pull/430/commits/af54a3ac187878869e28041f5a58a8380eefc8e8

Comment (zhtmike) [reply mindocr/data/transforms/svtr_transform.py:252]: Yes, this should reduce some computation time, can be replace by #430 once it is merged

**Review by hadipash — APPROVED** [review]

File: configs/rec/svtr/README.md [hunk 2]
```diff
@@ -104,16 +121,14 @@ data_lmdb_release/
...
 └── validation
     ├── data.mdb
     └── lock.mdb
 ```
 
-#### 3.1.3 Dataset Usage
-
 Here we used the datasets under `training/` folders for training, and the union dataset `validation/` for validation. After training, we used the datasets under `evaluation/` to evaluate model accuracy.
 
 **Training:** (total 14,442,049 samples)
```
[the remark this answers is not in the document]
Comment (zhtmike) [reply configs/rec/svtr/README.md]: Right, fixed.

**Comment by SamitHuang** [comment]
Nice work. What leads to the FPS improved from 2968 to 4560?

**Comment by zhtmike** [comment]
> Nice work. What leads to the FPS improved from 2968 to 4560?

Not very sure. Seems removing some long text sample will be helpful. (CTC loss has max. length limit, too.long image will not contributed to the loss value but still cost some time in the previous setting)

**Review by HaoyangLee — APPROVED** [review]

File: mindocr/data/transforms/svtr_transform.py [hunk 3]
```diff
@@ -242,59 +239,66 @@ def __init__(self, factor=4, base_size=(128, 512)):
...
-        for _ in range(self.factor):
+        for _ in range(factor):
             scale_img = cv2.pyrDown(scale_img)
         scale_img = cv2.resize(scale_img, (src_w, src_h), interpolation=get_interpolation())
         return scale_img
 
 
 class CVGaussianNoise(object):
-    def __init__(self, mean=0, var=20):
+    def __init__(self, mean=0, varience=20):
```
[the remark this answers is not in the document]
Comment (zhtmike) [reply mindocr/data/transforms/svtr_transform.py]: Yes, fixed
